app/assets/src/components/visualizations/table/Table.jsx

```diff
@@ -49,21 +49,24 @@ class Table extends React.Component {
 
     const { sortBy, sortDirection } = this.state;
 
-    const columnSortFunction = (find({ dataKey: sortBy }, columns) || {})
-      .sortFunction;
-
-    const sortedData =
-      sortable && sortBy
-        ? orderBy(
-            [
-              columnSortFunction
-                ? row => columnSortFunction(row[sortBy])
-                : sortBy,
-            ],
-            [sortDirection === SortDirection.ASC ? "asc" : "desc"],
-            this.props.data
-          )
-        : data;
+    let sortedData = data;
+    if (sortable && sortBy) {
+      const sortColumn = find({ dataKey: sortBy }, columns) || {};
+      const sortDirectionStr =
+        sortDirection === SortDirection.ASC ? "asc" : "desc";
+      if (sortColumn.sortFunction) {
+        sortedData = sortColumn.sortFunction({
+          data,
+          sortDirection: sortDirectionStr,
+        });
+      } else {
+        sortedData = orderBy(
+          sortColumn.sortKey || sortBy,
+          [sortDirectionStr],
+          data
+        );
+      }
+    }
```

Comment:
Looks great!

Reply:
Maybe add some documentation around the parameters and outputs of `sortColumn.sortFunction` to the `propTypes` below.